x-pack/plugins/security_solution/server/lib/entity_analytics/utils/transforms.ts

```diff
@@ -119,50 +125,90 @@ export const startTransform = async ({
+      `Unable to find transform status for [${transformId}] while attempting to schedule`
     );
   }
-  if (hasTransformStarted(transformStats.transforms[0])) {
-    return;
-  }
 
-  return esClient.transform.startTransform({ transform_id: transformId });
+  if (!hasTransformStarted(transformStats.transforms[0])) {
+    await esClient.transform.startTransform({
+      transform_id: transformId,
+    });
+  } else {
+    await esClient.transform.scheduleNowTransform({
+      transform_id: transformId,
+    });
+  }
 };
 
-export const scheduleTransformNow = async ({
+/**
+ * Whenever we change the latest transform configuration, we must ensure we update the transform in environments where it has already been installed.
+ */
+const upgradeLatestTransformIfNeeded = async ({
   esClient,
-  transformId,
+  namespace,
+  logger,
 }: {
   esClient: ElasticsearchClient;
-  transformId: string;
+  namespace: string;
+  logger: Logger;
 }): Promise<TransformStartTransformResponse | void> => {
-  const transformStats = await esClient.transform.getTransformStats({
+  const transformId = getLatestTransformId(namespace);
+  const latestIndex = getRiskScoreLatestIndex(namespace);
+  const timeSeriesIndex = getRiskScoreTimeSeriesIndex(namespace);
+
+  const response = await esClient.transform.getTransform({
     transform_id: transformId,
   });
-  if (transformStats.count <= 0) {
-    throw new Error(
-      `Unable to find transform status for [${transformId}] while attempting to schedule now`
-    );
-  }
 
-  if (hasTransformStarted(transformStats.transforms[0])) {
-    await esClient.transform.scheduleNowTransform({
-      transform_id: transformId,
-    });
-  } else {
-    await esClient.transform.startTransform({
+  const newConfig = getTransformOptions({
+    dest: latestIndex,
+    source: [timeSeriesIndex],
+  });
+
+  if (isTransformOutdated(response.transforms[0], newConfig)) {
+    logger.info(`Upgrading transform ${transformId}`);
+
+    const { latest: _unused, ...changes } = newConfig;
+
+    await esClient.transform.updateTransform({
       transform_id: transformId,
+      ...changes,
     });
   }
 };
 
 export const scheduleLatestTransformNow = async ({
   namespace,
   esClient,
+  logger,
 }: {
   namespace: string;
   esClient: ElasticsearchClient;
+  logger: Logger;
 }): Promise<void> => {
   const transformId = getLatestTransformId(namespace);
+
+  try {
+    await upgradeLatestTransformIfNeeded({ esClient, namespace, logger });
+  } catch (err) {
+    logger.error(
+      `There was an error upgrading the transform ${transformId}. Continuing with transform scheduling. ${err.message}`
+    );
+  }
+
   await scheduleTransformNow({ esClient, transformId });
 };
+
+/**
+ * Whitelist the transform fields that we can update.
+ */
+const UPDATED_TRANSFORM_PROPERTIES = ['sync', 'settings', 'frequency', 'dest', 'source.index'];
+
+const isTransformOutdated = (
```

Comment:
Transforms have a `_meta` property for metadata, in fleet we sometimes used this to hold a version, I propose that we have something like:

```
  version: 1, // no version would mean it is version 0, going forward when we change the schema we will change the version, we should have a unit test to force this
  managed_by: 'security-entity-analytics', // if we do set it as managed we should add this field
  managed: true; // do we set it as managed currently? We probably should be.
}
```

Reply:
Done! Can you take a look?